Line/android/LINELoginDemo/app/src/main/java/com/google/firebase/linelogindemo/activity/MainActivity.java

```diff
@@ -0,0 +1,151 @@
+import android.support.v7.app.AppCompatActivity;
+import android.os.Bundle;
+import android.util.Log;
+import android.view.View;
+import android.widget.Button;
+import android.widget.TextView;
+
+import com.android.volley.toolbox.ImageLoader;
+import com.android.volley.toolbox.NetworkImageView;
+import com.google.android.gms.tasks.OnFailureListener;
+import com.google.android.gms.tasks.OnSuccessListener;
+import com.google.firebase.auth.FirebaseAuth;
+import com.google.firebase.auth.FirebaseUser;
+import com.google.firebase.linelogindemo.R;
+import com.google.firebase.linelogindemo.util.LineLoginHelper;
+import com.google.firebase.linelogindemo.util.NetworkSingleton;
+
+public class MainActivity extends AppCompatActivity {
+
+    // Views
+    private Button mLineLoginButton;
+    private View mLoggedInView;
+    private TextView mDisplayNameText;
+    private NetworkImageView mProfileImageView;
+
+    // Helpers
+    private ImageLoader mImageLoader;
+    private LineLoginHelper mLineLoginHelper;
+
+    private final String TAG = this.getClass().getSimpleName();
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+        bindUIElements();
+
+        mImageLoader = NetworkSingleton.getInstance(this).getImageLoader();
+        mLineLoginHelper = new LineLoginHelper(this);
+        updateUI();
+    }
+
+    private void bindUIElements() {
+        // Bind views
+        mLineLoginButton = (Button) findViewById(R.id.line_login_button);
+        mLoggedInView = (View) findViewById(R.id.logged_in_view);
+        mDisplayNameText = (TextView) findViewById(R.id.display_name_text);
+        mProfileImageView = (NetworkImageView) findViewById(R.id.profile_image_view);
+
+        // Bind buttons' OnClickListener
+        mLineLoginButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                onTapLineLogin();
+            }
+        });
+        Button logoutButton = (Button) findViewById(R.id.logout_button);
+        logoutButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                onTapLogout();
+            }
+        });
+    }
+
+    void onTapLineLogin() {
+        // Show loading dialog
+        final ProgressDialog dialog = new ProgressDialog(this);
+        dialog.setMessage(getString(R.string.logging_in_using_line));
+        dialog.setCancelable(false);
+        dialog.show();
+
+        // Kick start login progress
+        mLineLoginHelper
+                .startLineLogin()
+                .addOnSuccessListener(new OnSuccessListener<FirebaseUser>() {
+                    @Override
+                    public void onSuccess(FirebaseUser firebaseUser) {
+                        Log.d(TAG, "LINE Login was successful.");
+
+                        updateUI();
```

Comment:
Fixed in this [commit](https://github.com/firebase/custom-auth-samples/pull/1/commits/fd56eda6713e7fd016d5c8ad820c78d2ffda97bf) :)